web/modules/custom/ct_user/src/Controller/GetUserPatchesController.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Drupal\ct_user\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use DateTime;
+use DateTimeZone;
+use Drupal\Core\Datetime\DrupalDateTime;
+
+/**
+ * A controller to get daily patches count for the last 1 year for each user.
+ */
+class GetUserPatchesController extends ControllerBase {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs a new GetUserPatchesController object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * Returns a JSON response with daily patches count for the last 1 year.
+   *
+   * @param int $current_user_id
+   *   The user ID for which you want to retrieve patches.
+   *
+   * @return \Symfony\Component\HttpFoundation\JsonResponse
+   *   The JSON response.
+   */
+  public function content($current_user_id) {
+    $days = 366;
+    $data = [];
+    $key = 0;
+    $current_date = date('Y-m-d');
+
+    while ($days > 0) {
```

Comment:
You are essentially running this query 366 times. Can you instead write one query to get all this data using aggregation? I understand that this might be difficult with entity query and we might have to use SQL directly.